gpiozero/pins/__init__.py

```diff
@@ -205,6 +312,26 @@ def _set_bounce(self, value):
         detection, measured in seconds. If bounce detection is not currently in
         use, this is ``None``.
 
+        For example, if :attr:`edge` is currently "rising", :attr:`bounce` is
+        currently 5/1000 (5ms), then the waveform below will only fire
+        :attr:`when_changed` on two occasions despite there being three rising
+        edges:
+
+        .. code-block:: text
+
+            TIME 0...1...2...3...4...5...6...7...8...9...10..11..12 ms
+
+            bounce elimination   |===================| |==============
+
+            HIGH - - - - >       ,--. ,--------------. ,--.
+                                 |  | |              | |  |
+                                 |  | |              | |  |
+            LOW  ----------------'  `-'              `-'  `-----------
+                                 :                     :
+                                 :                     :
+                           when_changed          when_changed
+                               fires                 fires
+
```

Comment:
Dunno, it's not something I've ever looked at in detail. I guess one way of doing it would be to only report state-changes when the state has remained constant for a specific time, but then you obviously lose your 'sharp' edge-detection (which I guess implies that it'd be handled by a separate attribute entirely).